packages/version_repository_interface/lib/src/models/version.dart

```diff
@@ -0,0 +1,63 @@
+import "package:version_repository_interface/src/models/compatibiliy.dart";
+import "package:version_repository_interface/src/utils/version_string.dart";
+
+/// A version.
+class Version {
+  /// Parse a version from a string.
+  factory Version.parse({required String version}) {
+    if (!version.isSemanticVersion()) {
+      throw Exception("Invalid version format");
+    }
+
+    var versionParts = version.versionParts;
+
+    return Version._(
+      major: versionParts[0],
+      minor: versionParts[1],
+      patch: versionParts[2],
+    );
+  }
+
+  /// Create a new version.
+  const Version._({
+    required this.major,
+    required this.minor,
+    required this.patch,
+  });
+
+  /// The major version.
+  final String major;
+
+  /// The minor version.
+  final String minor;
+
+  /// The patch version.
+  final String patch;
+
+  @override
+  String toString() => "$major.$minor.$patch";
+
+  /// Compare this version to another version.
+  /// Returns the compatibility of the versions.
+  VersionCompatibiliy compare(Version otherVersion) {
```

Comment:
Could we have 1 simple test for this method in the tests folder

Reply:
no